In [1]:
import pandas as pd
import numpy as np

In [2]:
file = "purchase_data.csv"
file_df = pd.read_csv(file)
file_df.head()
#this opens the csv file and shows us the headings of the csv file and the first 5 rows. 

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [45]:
#Player Count - Total Number of Players

df_result = pd.DataFrame(index=[0], columns=['Total Number of Players'])

df_result['Total Number of Players'][0] = len(file_df['SN'].unique())

df_result.head()

Total Number of Players
0                     576

In [96]:
#Purchasing Analysis
#Number of Unique Items (How many purchases of individual items, not necessarily "unique" to the game)
unique_items_count = len(file_df['Item Name'].unique())

#total number of purchases
total_number_count = len(file_df)
#total Revenue
total_revenue = sum(file_df['Price'])

#average purchase price
average_purchase_price = total_revenue / total_number_count

data = {'Number of Unique Items': unique_items_count, 
        'Total Number of Purchases': total_number_count, 'Total Revenue': total_revenue, 
        'Average Purchase Price': average_purchase_price
       }

df_result = pd.DataFrame(index=[0], data=data)

#result.columns = ['Number of Unique Items', 'Average Purchase Price', 'Total Number of Purchases', 'Total Revenue']

df_result.head()

#print(unique_items_count, average_purchase_price, total_number_count, total_revenue)


Number of Unique Items  Total Number of Purchases  Total Revenue  \
0                     179                        780        2379.77   

   Average Purchase Price  
0                3.050987

In [94]:
#Gender Demographics
#percentage count of male players
#Percentage count of female players
#percentage count of other/non-disclosed

df_result = pd.DataFrame(index=file_df['Gender'].unique())
df_result['Player Count'] = file_df.drop_duplicates('SN').groupby('Gender').size()
df_result['Percentage'] = file_df.drop_duplicates('SN').groupby('Gender').size() / len(file_df['SN'].unique()) * 100.0

df_result.round(2).head()


Player Count  Percentage
Male                            484       84.03
Other / Non-Disclosed            11        1.91
Female                           81       14.06

In [99]:
#Purchasing Analysis (Gender)
#The below - broken down by gender

result = file_df[['Price', 'Gender']].groupby('Gender').agg(['sum', 'mean', 'size'])['Price']

#Purchase Count (M, F)
#Average Purchase Price (M, F)
#Total Purchase Value (M, F)
#Average Purchase Total per Person (M,F)

sum_price = file_df.groupby('Gender')['Price'].sum()
count_each = file_df.groupby('Gender')['SN'].nunique()
average_purchase_per_person = sum_price / count_each

result['Avg Total Purchase per Person'] = average_purchase_per_person

result.columns = ['size', 'mean', 'sum', 'Avg Total Purchase per Person']

result.rename(columns={'size': 'Purchase Count', 'mean': 'Average Purchase Price', 'sum': 'Total Purchase Value'}, inplace=True)

result.round(2).head()


Purchase Count  Average Purchase Price  \
Gender                                                          
Female                         361.94                    3.20   
Male                          1967.64                    3.02   
Other / Non-Disclosed           50.19                    3.35   

                       Total Purchase Value  Avg Total Purchase per Person  
Gender                                                                      
Female                                  113                           4.47  
Male                                    652                           4.07  
Other / Non-Disclosed                    15                           4.56

In [92]:

def get_age_group(age):
    a = [9, 14, 19, 24, 29, 34, 39, 150]
    g = ['<10', '10-14', '15-19', '20-24', '25-29', '30-34', '35-39', '40+']
    
    for i in range(len(a)):
        if age <= a[i]:
            return g[i]
    
    #return "ERROR"


file_df['age_group'] = file_df['Age'].apply(lambda age : get_age_group(age))
        

df_result = pd.DataFrame(index=file_df['age_group'].unique())
df_result['Count'] = file_df.drop_duplicates('SN').groupby('age_group').size()
df_result['Percentage'] = file_df.drop_duplicates('SN').groupby('age_group').size() / len(file_df['SN'].unique()) * 100.0

df_result.round(2).head(15)


Count  Percentage
20-24    258       44.79
40+       12        2.08
35-39     31        5.38
30-34     52        9.03
25-29     77       13.37
10-14     22        3.82
<10       17        2.95
15-19    107       18.58

In [75]:
#age demographics
#broken down into bins of 4 years (<10, 10-14, 15-19, 20-max)

#Purchase count
#average purchase price
#total purchase value 
#average purchase total per person by age group

result = file_df[['Price', 'age_group']].groupby('age_group').agg(['sum', 'mean', 'size'])['Price']

#Total Purchase Value (M, F)

#Average Purchase Total per Person (M,F)

sum_price = file_df.groupby('age_group')['Price'].sum()
#Purchase Count (M, F)
count_each = file_df.groupby('age_group')['SN'].nunique()
#Average Purchase Price (M, F)
average_purchase_per_person = sum_price / count_each

result['Avg Total Purchase per Person'] = average_purchase_per_person

result.columns = ['size', 'mean', 'sum', 'Avg Total Purchase per Person']

result.rename(columns={'size': 'Total Purchase Value', 'mean': 'Average Purchase Price', 'sum': 'Purchase Count'}, inplace=True)

result.round(2).head(15)

Total Purchase Value  Average Purchase Price  Purchase Count  \
age_group                                                                 
10-14                     82.78                    2.96              28   
15-19                    412.89                    3.04             136   
20-24                   1114.06                    3.05             365   
25-29                    293.00                    2.90             101   
30-34                    214.00                    2.93              73   
35-39                    147.67                    3.60              41   
40+                       38.24                    2.94              13   
<10                       77.13                    3.35              23   

           Avg Total Purchase per Person  
age_group                                 
10-14                               3.76  
15-19                               3.86  
20-24                               4.32  
25-29                               3.81  
30-34                               4.12  
35-39                               4.76  
40+                                 3.19  
<10                                 4.54

In [83]:
#Top Spenders
#Identify the top 5 spenders in the game by total purchase value, then list (in a table)

result = file_df[['SN', 'Price']].groupby('SN').agg(['size', 'sum', 'mean'])['Price']

#print(result.columns)

result.rename(columns={'size': 'Purchase Count', 'mean': 'Average Purchase Price', 'sum': 'Total Purchase Value'}, inplace=True)

result.sort_values(by='Total Purchase Value', ascending=False).round(2).head()

#SN
#Purchase Count 
#Average Purchase Price
#Total Purchase Value

Purchase Count  Total Purchase Value  Average Purchase Price
SN                                                                       
Lisosia93                 5                 18.96                    3.79
Idastidru52               4                 15.45                    3.86
Chamjask73                3                 13.83                    4.61
Iral74                    4                 13.62                    3.40
Iskadarya95               3                 13.10                    4.37

In [86]:
#Most Popular Items
#Identify the 5 most popular items by purchase count, then list (in a table)

#item id
#item name
#item price

result = file_df.groupby(['Item Name', 'Item ID']).agg(['size', 'sum'])['Price']
result['Price'] = result['sum'] / result['size']

result.rename(columns={'size': 'Purchase Count', 'sum': 'Total Purchase Value'}, inplace=True)

result.sort_values(by='Purchase Count', ascending=False).head()

#purchase count 
#total purchase value

Purchase Count  \
Item Name                                    Item ID                   
Oathbreaker, Last Hope of the Breaking Storm 178                  12   
Extraction, Quickblade Of Trembling Hands    108                   9   
Nirvana                                      82                    9   
Fiery Glass Crusader                         145                   9   
Pursuit, Cudgel of Necromancy                19                    8   

                                                      Total Purchase Value  \
Item Name                                    Item ID                         
Oathbreaker, Last Hope of the Breaking Storm 178                     50.76   
Extraction, Quickblade Of Trembling Hands    108                     31.77   
Nirvana                                      82                      44.10   
Fiery Glass Crusader                         145                     41.22   
Pursuit, Cudgel of Necromancy                19                       8.16   

                                                      Price  
Item Name                                    Item ID         
Oathbreaker, Last Hope of the Breaking Storm 178       4.23  
Extraction, Quickblade Of Trembling Hands    108       3.53  
Nirvana                                      82        4.90  
Fiery Glass Crusader                         145       4.58  
Pursuit, Cudgel of Necromancy                19        1.02

In [103]:
#Most profitable items
#identify the 5 most revenue-generating items by total purchase value, then list (in a table)

#item id
#item name 
#item price

result = file_df.groupby(['Item Name', 'Item ID']).agg(['size', 'sum'])['Price']
result['Price'] = result['sum'] / result['size']

result.rename(columns={'size': 'Purchase Count', 'sum': 'Total Purchase Value'}, inplace=True)

result.sort_values(by='Total Purchase Value', ascending=False).head()

#result.sort_values(by='Total Purchase Value', ascending=False).head(20)
#purchase count 
#total Purchase Value

Purchase Count  \
Item Name                                    Item ID                   
Oathbreaker, Last Hope of the Breaking Storm 178                  12   
Nirvana                                      82                    9   
Fiery Glass Crusader                         145                   9   
Final Critic                                 92                    8   
Singed Scalpel                               103                   8   

                                                      Total Purchase Value  \
Item Name                                    Item ID                         
Oathbreaker, Last Hope of the Breaking Storm 178                     50.76   
Nirvana                                      82                      44.10   
Fiery Glass Crusader                         145                     41.22   
Final Critic                                 92                      39.04   
Singed Scalpel                               103                     34.80   

                                                      Price  
Item Name                                    Item ID         
Oathbreaker, Last Hope of the Breaking Storm 178       4.23  
Nirvana                                      82        4.90  
Fiery Glass Crusader                         145       4.58  
Final Critic                                 92        4.88  
Singed Scalpel                               103       4.35

#Observations 
#Of the 576 total players, 484 (84%) of them are male.  
#The game skews toward players in the 20-24 range, with 44.79% of the players being in this age range. 58% are in their 20's.
#Players in the 20-24 range are more likely to make a purchase of a game enhancing item, and spend more money in the game than other age groups. I would like to see the purchase breakdown of age groups by Gender to see which age group by gender spends the most money. 